espree.js

```diff
@@ -3535,6 +3539,7 @@ function parseAssignmentExpression() {
         }
 
         if (params) {
+            state.parenthesisCount--;
```

Comment:
This was the fix for the nested default params issue (`(a, b=(c)=>c) => {}`). All of the tests pass yes, but I'm not super duper confident I understand how the `parenthsisCount` stuff works. What I'm trying to say here is, if you're inside an arrow function param let's not worry about 1 set of parens when it comes to determining if other things are arrow functions, etc.